modules/s3-bucket/main.tf

```
@@ -57,11 +87,11 @@ data "aws_iam_policy_document" "custom_policy" {
     actions = var.custom_policy_actions
 
     resources = [
-      format("arn:aws:s3:::%s-%s-%s-%s", module.this.namespace, module.this.environment, module.this.stage, module.this.name),
-      format("arn:aws:s3:::%s-%s-%s-%s/*", module.this.namespace, module.this.environment, module.this.stage, module.this.name)
+      format("arn:%s:s3:::%s", local.aws_partition, module.this.id),
+      format("arn:%s:s3:::%s/*", local.aws_partition, module.this.id)
```

We should not be constructing ARNs like this. Use
```suggestion
      module.s3_bucket.bucket_id,
      "${module.s3_bucket.bucket_id}/*",
```